lib/amazon.js

```diff
@@ -39,11 +42,15 @@ function addPermission ({ lambdaArn, restApiId }) {
   return lambda.addPermission(params).promise()
 }
 
-exports.createFunction = function ({ zipFile, functionName, role, environment, onFailedAttempt }) {
-  verifyZipFile(zipFile)
+exports.createFunction = function ({ zipFile, functionName, role, environment, onFailedAttempt, bucket, s3Key }) {
+  verifyZipFile(zipFile, { sizeLimit: bucket ? null : MAX_LAMBDA_ZIP_SIZE })
 
   const params = {
-    Code: { ZipFile: zipFile },
+    Code: {
+      ZipFile: !bucket ? zipFile : null,
```

Comment:
Maybe something like:

```js
const code = bucket ? { S3Key: s3Key, S3Bucket: bucket } : { ZipFile: zipFile }
```

_side note:_ oh god why are they using AwsCase in their JavaScript API?! 😭 😭 🙏